mergekit/plan.py

```diff
@@ -246,31 +250,13 @@ def plan_to_disk(self, out_path: str) -> List[Task]:
         """Plan the merge to be streamed to disk, returning a list of tasks."""
         self._plan()
 
-        writer_task = TensorWriterTask(
-            out_path=out_path,
-            max_shard_size=self.options.out_shard_size,
-            safe_serialization=self.options.safe_serialization,
-        )
         save_tasks = []
         for weight, tensor_task in self._tensors:
             save_tasks.append(
-                SaveTensor(
-                    tensor_name=weight.name,
-                    tensor_task=tensor_task,
-                    writer_task=writer_task,
-                    clone=self.options.clone_tensors,
-                    optional=weight.optional,
-                    dtype=weight.force_dtype,
-                )
+                tensor_task
             )
-        finalize = FinalizeModel(
```

Comment:
Totally fine to not do the `finalize` task when we're doing metrics, but this is needed for merges - I think as is this makes merges not write out correctly.